crypto/bn/build.info

```diff
@@ -7,6 +7,17 @@ SOURCE[../../libcrypto]=\
         bn_recp.c bn_mont.c bn_mpi.c bn_exp2.c bn_gf2m.c bn_nist.c \
         bn_depr.c bn_const.c bn_x931p.c bn_intern.c bn_dh.c bn_srp.c \
         bn_rsa_fips186_4.c
+
+SOURCE[../../providers/fips]=\
+        bn_add.c bn_div.c bn_exp.c bn_lib.c bn_ctx.c bn_mul.c bn_mod.c \
+        bn_print.c bn_rand.c bn_shift.c bn_word.c bn_blind.c \
+        bn_kron.c bn_sqrt.c bn_gcd.c bn_prime.c bn_sqr.c \
+        {- $target{bn_asm_src} -} \
+        bn_recp.c bn_mont.c bn_mpi.c bn_exp2.c bn_gf2m.c bn_nist.c \
+        bn_const.c bn_x931p.c bn_intern.c bn_dh.c \
+        bn_rsa_fips186_4.c
```

Comment:
Aren't you getting tired of the repetitions?  (I've seen that quite a number of times now)

Personally, I use perl templating, like this:

```
{- our @src = ( qw( bn_add.c bn_div.c bn_exp.c bn_lib.c bn_ctx.c bn_mul.c bn_mod.c
                    bn_print.c bn_rand.c bn_shift.c bn_word.c bn_blind.c
                    bn_kron.c bn_sqrt.c bn_gcd.c bn_prime.c bn_sqr.c
                    bn_recp.c bn_mont.c bn_mpi.c bn_exp2.c bn_gf2m.c bn_nist.c
                    bn_const.c bn_x931p.c bn_intern.c bn_dh.c
                    bn_rsa_fips186_4.c ), $target{bn_asm_src} ); -}
LIBS=../../libcrypto
SOURCE[../../libcrypto]={- join(' ', @src) -} bn_err.c bn_depr.c bn_srp.c
SOURCE[../../providers/fips]={- join(' ', @src) -}
```

But maybe this is a good reason to have local variables in build.info files...

```
$COMMONSRC=\
        bn_add.c bn_div.c bn_exp.c bn_lib.c bn_ctx.c bn_mul.c bn_mod.c \
        bn_print.c bn_rand.c bn_shift.c bn_word.c bn_blind.c \
        bn_kron.c bn_sqrt.c bn_gcd.c bn_prime.c bn_sqr.c \
        {- $target{bn_asm_src} -} \
        bn_recp.c bn_mont.c bn_mpi.c bn_exp2.c bn_gf2m.c bn_nist.c \
        bn_const.c bn_x931p.c bn_intern.c bn_dh.c \
        bn_rsa_fips186_4.c
LIBS=../../libcrypto
SOURCE[../../libcrypto]=$COMMONSRC bn_err.c bn_depr.c bn_srp.c
SOURCE[../../providers/fips]=$COMMONSRC
```

Would that be helpful?

Reply:
I went with your suggestion. I'm not sure whether having local variables in the build.info files might be more than is needed.

Reply:
Depends on how much work is involved - If it is  easy to add, it certainly looks a lot cleaner.

Reply:
It shouldn't be too hard to add.  It's a fairly simple string replace scheme.

Reply:
In that case...sure. Why not?